misk-redis/src/main/kotlin/misk/redis/Redis.kt

```diff
@@ -800,3 +800,20 @@ inline fun checkHrandFieldCount(count: Long) {
   }
 }
 
+internal fun getSlotErrorOrNull(op: String, keys: List<ByteArray>): Throwable? {
+  val slots = keys.map { JedisClusterCRC16.getSlot(it) }.distinct()
```

Comment:
Do we need this logic client-side? I would imagine that Redis would throw a `CROSSSLOT` for us

Reply:
Non-blocking bc I see that this check isn't entirely net-new

Reply:
Sadly it doesn't behave as expected in pipelines unless we do the client-side check. I would prefer a more consistent experience across the client, which is why I promoted it to the non-pipelined operations that might encounter cross slot errors as well.